doc/src/guides/solving/solve-diophantine-equation.md

```diff
@@ -0,0 +1,170 @@
+# Solve a Diophantine Equation Algebraically
+
+Use SymPy to solve a Diophantine equation (find integer solutions to a
+polynomial equation) algebraically, returning a parameterized general solution
+if possible. For example, solving the [Pythagorean
+theorem](https://en.wikipedia.org/wiki/Pythagorean_theorem) $a^2 + b^2 = c^2$
+yields $(a=2pq, b=p^2-q^2, c=p^2-q^2)$.
+
+## Alternatives to Consider
+
+There are few alternatives for finding a parameterized general solution a
+Diophantine equation. {func}`~.solve` simply solves for one variable in terms of
+the others. For example, attempting to solve $a^2 + b^2 = c^2$ for $a$, $b$, and
+$c$ can only reveal that $a = \pm \sqrt{c^2-b^2}$:
+
+```py
+>>> from sympy import solve
+>>> from sympy import symbols
+>>> a, b, c = symbols("a, b, c", integer=True)
+>>> solve(a**2 + b**2 - c**2, [a, b, c], dict=True)
+[{a: -sqrt(-b**2 + c**2)}, {a: sqrt(-b**2 + c**2)}]
+```
```

Comment:
I'm debating whether to remove this code block--I'm concerned having the first code block be of a function that's not the subject one, `diophantine`, will confuse visitors.

Reply:
That may be prudent. However, if you do keep it, it should be indented so that it aligns with the bullet point.